lib/puppet/provider/ec2_instance/v2.rb

```diff
@@ -70,6 +71,7 @@ def self.instance_to_hash(region, instance)
       tags: tags,
       region: region,
       hypervisor: instance.hypervisor,
+      iam_instance_profile_arn: instance.iam_instance_profile ? instance.iam_instance_profile.arn : nil,
```

Comment:
Mmm, so it might make sense here for the `iam_instance_profile_arn` to be a property (as you have it) and the `iam_instance_profile_name` to be a param (ie. it's only passed in on create). That would mean we don't need it back from the API. I think at least to begin with that's good enough.